lib/rubocop/cop/lint/non_local_exit_from_iterator.rb

```diff
@@ -14,7 +14,10 @@ module Lint
       # * the return is not contained in an inner scope, e.g. a lambda or a
       # method definition.
       #
-      # @example
+      # Setting `AllowReturnValue` to `false` overrides the default behavior
+      # for the first condition.
+      #
+      # @example AllowReturnValue: true (default)
```

Comment:
@Earlopain there are a lot of rules here -- in my mind, a setting called `Strict` should turn off all of the exceptions. That's why I went for a more narrow setting name to disable just "No value is returned".

I just ran a version of that "strictest" setting against my ~5000 line ruby project and I think it probably helps with correctness. I got about 25 new errors there: mostly around "returning from a block that doesn't have any arguments".

This does change the meaning of the Cop, a bit: it's no longer "non local exit from iterator" but "you should `break` instead of `return` from a block". Do you think that makes it worthwhile to add an entirely different Cop?